include/deal.II/sundials/arkode.h

```diff
@@ -907,6 +1372,115 @@ namespace SUNDIALS
 #  endif   // DEAL_II_WITH_PETSC
   };
 
+#  if DEAL_II_SUNDIALS_VERSION_GTE(4, 0, 0)
+
+  /**
+   * A linear operator that wraps SUNDIALS functionality.
+   */
+  template <typename VectorType>
+  struct SundialsOperator
+  {
+    /**
+     * Apply this LinearOperator to @p src and store the result in @p dst.
+     */
+    void
+    vmult(VectorType &dst, const VectorType &src) const;
+
+    /**
+     * Constructor.
+     *
+     * @param solver the ARKode solver that uses this operator
+     * @param A_data data required by @p a_times_fn
+     * @param a_times_fn a function pointer to the function that computes A*v
+     */
+    SundialsOperator(ARKode<VectorType> &solver,
+                     void *              A_data,
+                     ATimesFn            a_times_fn);
+
+  private:
+    /**
+     * Reference to the ARKode object that uses this SundialsOperator.
+     */
+    ARKode<VectorType> &solver;
+    /**
+     * Data necessary to evaluate a_times_fn.
+     */
+    void *A_data;
+
+    /**
+     * Function pointer declared by SUNDIALS to evaluate the matrix vector
+     * product.
+     */
+    ATimesFn a_times_fn;
+  };
+
```

Comment:
Would you mind to add two more empty lines here? We strive to have three empty lines between different objects or between implementations, see https://www.dealii.org/developer/doxygen/deal.II/CodingConventions.html (item 4 in the list).